graylog2-web-interface/src/views/components/visualizations/bar/BarVisualization.jsx

```diff
@@ -42,10 +43,16 @@ const BarVisualization: VisualizationComponent = ({ config, data, effectiveTimer
   const _seriesGenerator = (type, name, labels, values): ChartDefinition => ({ type, name, x: labels, y: values, opacity });
 
   const rows = data.chart || Object.values(data)[0];
+  const chartDataResult = chartData(config, rows, 'bar', _seriesGenerator);
+  if (config.eventAnnotation && data.events) {
+    const { eventChartData, shapes } = EventHandler.toVisualizationData(data.events, config.formattingSettings);
+    chartDataResult.push(eventChartData);
+    layout.shapes = shapes;
+  }
```

Comment:
I don't know the best place for it right now, but it would be great to extract this logic to a function, which we can use in all three charts.

Reply:
`toVisualizationData` is already all that I could extract. If you have any better idea I am open for suggestions.